modules/tensor_mechanics/src/actions/CommonLineElementAction.C

```diff
@@ -0,0 +1,26 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CommonLineElementAction.h"
+#include "LineElementAction.h"
+
+registerMooseAction("TensorMechanicsApp", CommonLineElementAction, "meta_action");
+
+template <>
+InputParameters
+validParams<CommonLineElementAction>()
+{
+  InputParameters params = validParams<LineElementAction>();
```

Comment:
OK, so here you are actually grabbing the children type validParams. I really think that the common should always just directly apply to all children: https://github.com/idaholab/moose/blob/devel/framework/include/utils/InputParameters.h#L568-L584

Reply:
@permcody, I tried modifying the CommonLineElementAction using the steps above. But I am able to pass the error checks even after I move the error checks to LineElementAction::act(). Do I need to use some other function to check for error in this case instead of isParamValid? 

Also in commonOutputAction, the common parameters are set to the OutputWarehouse and then AddOutputAction sets the common parameters from the OutputWarehouse. I am not sure how it is different from what I am doing in LineElementAction (lines 236 - 239) where I am getting the parameters from the CommonLineElementAction and setting them to LineElementAction parameters.

Reply:
Thanks for the clarification, @permcody . Yes, I am interested in setting the parameters in a parent block. So, I am modeling the CommonLineElementAction similar to the CommonTensorMechanicsAction. So, I will leave the design as it is for now and change it later when this capability is available in the MOOSE parser.